fpsim/wrappers.py

```diff
@@ -0,0 +1,1216 @@
+
+**Method name shortcuts:**
+- 'pill' = Pills
+- 'inj' = Injectables (DMPA, Sayana Press, etc.)
+- 'impl' = Implants (Jadelle, Implanon, etc.)
+- 'iud' = IUDs
+- 'cond' = Male condoms
+- 'btl' = Female sterilization (tubal ligation)
+- 'wdraw' = Withdrawal
+- 'othtrad' = Other traditional methods
+- 'othmod' = Other modern methods
+"""
+# endregion: Introduction
+
+import json
+from contextlib import contextmanager
+from dataclasses import dataclass, field
+from enum import Enum
+from functools import wraps
+from operator import attrgetter
+from pathlib import Path
+from typing import Dict, Optional, Sequence, Union, TypedDict, List, Any
+
+import numpy as np
+import sciris as sc
+
+from . import interventions as fpi
+from .methods import NewMethodConfig
+
+__all__ = [
+    'MethodIntervention',
+    'make_update_methods',
+    'MethodName',
+]
+
+# ==========================================
+# Enums and Type Definitions
+# ==========================================
+class MethodName(str, Enum):
+    """Enumeration of standard contraceptive method names."""
+    NONE = 'none'
+    PILL = 'pill'
+    IUD = 'iud'
+    INJECTABLE = 'inj'
+    CONDOM = 'cond'
+    BTL = 'btl'
+    WITHDRAWAL = 'wdraw'
+    IMPLANT = 'impl'
+    OTHER_TRAD = 'othtrad'
+    OTHER_MOD = 'othmod'
+    
+    @classmethod
+    def values(cls) -> List[str]:
+        """Return list of all method name values."""
+        return [m.value for m in cls]
+    
+    @classmethod
+    def excluding_none(cls) -> List[str]:
+        """Return list of method names excluding 'none'."""
+        return [m.value for m in cls if m != cls.NONE]
+
+class PreviewDict(TypedDict, total=False):
+    """Typed dictionary for preview() return value."""
+    year: float
+    efficacy: Optional[Dict[str, float]]
+    duration_months: Optional[Dict[str, Union[int, float]]]
+    p_use: Optional[float]
+    method_mix: Optional[Dict[str, float]]
+    switching_matrix: Optional[str]
+    new_method: Optional[Dict[str, Any]]
+    label: Optional[str]
+
+@dataclass(slots=True)
+class InterventionConfig:
+    """Container for intervention parameters with default values."""
+    eff: Dict[str, float] = field(default_factory=dict)
+    dur: Dict[str, Union[int, float]] = field(default_factory=dict)
+    p_use: Optional[float] = None
+    mix_values: Dict[str, float] = field(default_factory=dict)
+    method_mix_base: Optional[np.ndarray] = None
+    switch: Optional[dict] = None
+    new_method: Optional[NewMethodConfig] = None
+
+def load_config(source: Union[str, Path, dict]) -> dict:
+    """Load configuration from dict or JSON file."""
+    if isinstance(source, dict):
+        return sc.dcp(source)
+    
+    path = Path(source) if isinstance(source, str) else source
+    
+    if path.suffix != '.json':
+        raise ValueError(f'Unsupported file type: {path.suffix}. Expected .json')
+    
+    return json.loads(path.read_text())
+
+@contextmanager
+def _print_if(condition: bool, message: str):
```

Comment:
Minor comment, but seems like this should live in Starsim (if generally useful) or fp.utils